src/control/positionControl/pathPlanning/NumTreesPlanning.cpp

```diff
@@ -0,0 +1,200 @@
+                return backTrackPath(point);
+            }
+            // if we reach a halfway point, update the target to the final target again and push this new point to queue
+            if (point->isCollision(point->currentTarget, 0.1)) {
+                point->currentTarget = targetPosition;
+                pathQueue.pop();
+                pathQueue.push(point);
+                break;  // break from current while loop, we start looking at different branches again
+            }
+            // if we have collided with an obstacle; backtrack and branch to both sides using new intermediary points
+            if (collisionDetector.isCollisionBetweenPoints(point->parent->pos, point->pos)) {
+                int collisions = ++point->collisions;
+                pathQueue.pop();
+
+                std::pair<std::vector<Vector2>, PathPointer> newTargetsAndBranch = getNewTargets(point);
+                std::vector<Vector2> newTargets = newTargetsAndBranch.first;
+                PathPointer newBranchStart = newTargetsAndBranch.second;
+
+                // both left and right targets for now
+                for (const auto &newTarget : newTargets) {
+                    if (newBranchStart->branchHasTarget(newTarget)) continue;
+                    // compute new point and add it to branch
+                    PathPointer branch = computeNewPoint(targetPosition, newBranchStart, newTarget);
+                    branch->collisions = collisions;
+                    newBranchStart->addChild(branch);
+                    pathQueue.push(branch);
+                }
+                break;  // break from current while loop, we start looking at different branches again
+            }
+        }
+    }
+    if (interface::Output::showDebugNumTreeInfo()) {
+        std::cout << "ROBOT "
+                  << ": reached end of while loop, no path found" << std::endl;
+    }
+    return {};
+}
+
+/// calculate the remaining pathlength using straigth lines from current position to a posititon halfway and from
+/// halfway to the final position
+inline double NumTreesPlanning::remainingStraightLinePathLength(const Vector2 &currentPos, const Vector2 &halfwayPos, const Vector2 &finalPos) {
+    return (abs((halfwayPos - finalPos).length() + (currentPos - halfwayPos).length()));
+}
+
+/// backTracks the path from endPoint until it hits root and outputs in order from root->endPoint
+std::vector<Vector2> NumTreesPlanning::backTrackPath(PathPointer point) {
+    // backtrack the whole path till it hits the root node and return the vector of PathPoints
+    std::vector<Vector2> backTrackedPath = {};
+    while (point) {
+        backTrackedPath.push_back(point->pos);
+        if (point->parent == nullptr) {
+            break;
+        }
+        point.swap(point->parent);
+    }
+
+    // as older points get pushed back, the vector has to be reversed
+    std::reverse(backTrackedPath.begin(), backTrackedPath.end());
```

Comment:
Can we not prevent this?

Reply:
The alternative would be to insert at the front of the vector, which is not really efficient